fedimint-server/src/config/api.rs

```diff
@@ -185,21 +186,46 @@ impl ConfigGenApi {
     /// Once configs are generated, starts DKG and await its
     /// completion, calling a second time will return an error
     pub async fn run_dkg(&self) -> ApiResult<()> {
-        // Update our state to running DKG
-        let request = self.get_requested_params()?;
-        let response = self.get_consensus_config_gen_params(&request).await?;
-        let (params, registry) = {
+        let leader = {
             let mut state = self.require_status(ServerStatus::SharingConfigGenParams)?;
+            // Update our state
             state.status = ServerStatus::ReadyForConfigGen;
-            (
-                state.get_config_gen_params(&request, response.consensus)?,
-                state.settings.registry.clone(),
-            )
+            // Create a WSClient for the leader
+            state.local.clone().and_then(|local| {
```

Comment:
Afaik `state.local` can never be none in this state, should we `expect` here instead of `and_then`? Or under which conditions can it be `None`?